eventlet/support/greendns.py

```diff
@@ -178,16 +187,16 @@ def _readlines(self):
 
         Note that this performs disk I/O so can be blocking.
         """
-        lines = []
         try:
-            with open(self.fname, 'rU') as fp:
-                for line in fp:
-                    line = line.strip()
-                    if line and line[0] != '#':
-                        lines.append(line)
+            with open(self.fname, 'rb') as fp:
+                fdata = fp.read()
         except (IOError, OSError):
-            pass
-        return lines
+            return []
+
+        if not isinstance(fdata, str):  # Python 3
```

Comment:
2/3 type names and aliases get you confused. Think conceptually binary or text. You open file in binary mode regardless platform. Function must return text always. It means you have to `.decode()` always.

```
% python2.7                                                        :(
Python 2.7.10 (default, Jul 14 2015, 19:46:27)
>>> s = "text"
>>> type(s)
<type 'str'>
>>> s.decode('utf-8', 'ignore')
u'text'
% python3.6
Python 3.6.0 (default, Dec 24 2016, 08:02:28)
>>> s = b"text"
>>> type(s)
<class 'bytes'>
>>> s.decode('utf-8', 'ignore')
'text'
```